srcpkgs/gnupg/patches/0002-tests-Fix-tests-gpgme-for-in-source-tree-builds.patch

```diff
@@ -0,0 +1,63 @@
+From a4ec37b589a0646f2e756921f7abbb578cb5464d Mon Sep 17 00:00:00 2001
+From: NIIBE Yutaka <gniibe@fsij.org>
+Date: Wed, 21 Dec 2022 10:52:24 +0900
+Subject: [PATCH 2/2] tests: Fix tests/gpgme for in-source-tree builds.
+
+* tests/gpgme/Makefile.am: Don't use setup.scm/ dir.
+* tests/gpgme/all-tests.scm: Fix the name of the environment.
+
+--
+
+GnuPG-bug-id: 6313
+Fixes-commit: c19ea75f10d6278569619f90977ce7c820e9319d
+Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
+---
+ tests/gpgme/Makefile.am   | 5 ++---
+ tests/gpgme/all-tests.scm | 4 ++--
+ 2 files changed, 4 insertions(+), 5 deletions(-)
+
+diff --git a/tests/gpgme/Makefile.am b/tests/gpgme/Makefile.am
+index ca7be13df..ae98db4a0 100644
+--- a/tests/gpgme/Makefile.am
++++ b/tests/gpgme/Makefile.am
+@@ -47,8 +47,7 @@ check: xcheck
+ 
+ .PHONY: xcheck
+ xcheck:
+-	@$(MKDIR_P) setup.scm/tests \
+-	  tests/gpg lang/qt/tests lang/python/tests
++	@$(MKDIR_P) tests/gpg lang/qt/tests lang/python/tests
+ 	$(TESTS_ENVIRONMENT) $(abs_top_builddir)/tests/gpgscm/gpgscm$(EXEEXT) \
+ 	  $(abs_srcdir)/run-tests.scm $(TESTFLAGS) $(TESTS)
+ 
+@@ -61,4 +60,4 @@ CLEANFILES = *.log report.xml
+ all-local: $(required_pgms)
+ 
+ clean-local:
+-	-rm -rf setup.scm/tests tests/gpg lang/qt/tests lang/python/tests
++	-rm -rf tests lang
```

Comment:
If the failure because of in-source-tree build, you can simulated by force gnupg build out-of-tree. I don't think adding a patch and autotools is worth it.

Reply:
Feel free to work on building it out-of-tree, I think adding a patch that has been merged upstream and temporarily depend on autotools is worth it.

Reply:
@kruceter came in to save the day, and has provided a working patch for building out of tree. Not sure what I did wrong last night, but I couldn't get it to work then.